### DISCUSSION TITLE
Is there any paper of document about the theory detail of the model

### DISCUSSION THREAD

**Discussion by qinyuenlp** [snakers4/silero-vad#188] [Q&A] [answered]
Hi,
Thanks for sharing the code, it really help me a lot. Is there any paper or document can help me learn the theory detail of the model ?
When I use SlidingWindow method to feed audio data into your model, different window-size cause different VAD result. I want to know how the self._h and self._c in OnnxWrapper change exactly.

**Comment by snakers4** [comment, marked as answer]
Hi,

> Is there any paper or document can help me learn the theory detail of the model ?

There is no paper, but there is a short article - https://thegradient.pub/one-voice-detector-to-rule-them-all/

> When I use SlidingWindow method to feed audio data into your model, different window-size cause different VAD result. I want to know how the self._h and self._c in OnnxWrapper change exactly.

The results should be slightly different for different windows.
Please be careful and read the docstring in the utils.

https://github.com/snakers4/silero-vad/blob/ea7af70fe9b354364aa85a4245dded9448071e6a/utils_vad.py#L119-L171

i.e.:

```
     window_size_samples: int (default - 1536 samples) 
         Audio chunks of window_size_samples size are fed to the silero VAD model. 
         WARNING! Silero VAD models were trained using 512, 1024, 1536 samples for 16000 sample rate and 256, 512, 768 samples for 8000 sample rate. 
         Values other than these may affect model perfomance!! 
```

**Reply by snakers4** [reply]
@qinyuenlp

**Reply by qinyuenlp** [reply]
I mean, at `utils_vad.py` line 206 and 207, your way to feed chunk into model (assume window_size_samples=768) is
```
for current_start_sample in range(0, audio_length_samples, 768):
    chunk = audio[current_start_sample: current_start_sample + 768]
```
But for some reason, I try to feed as
```
for current_start_sample in range(0, audio_length_samples, 384):
    chunk = audio[current_start_sample: current_start_sample + 768]
```
They will feed same-size-chunk to model, but when I feed a "same" chunk like audio[3840:4608] by different slide-window-size into model, I got different result.

**Reply by snakers4** [reply]
I see, you are doing overlapped winbdows to increase quality / resolution?

The values should be similar, but they may differ because this is not really required, because the model has a built-in sequential inductive bias.

We used to use "overlapping windows" when our temporal resolution was bad, i.e. 100-200ms.

But now this is not required. How the model may behave unpredictably, because we did not train this model to handle overlapping windows.

Can you please plot the probablity charts for these two cases?

**Reply by qinyuenlp** [reply]
Sure, I'll plot that tomorrow, have no wav data in my own PC right now.
Thanks again.
